pennylane/ftqc/parametric_midmeasure.py

```diff
@@ -146,9 +395,141 @@ def label(self, decimals=None, base_label=None, cache=None):  # pylint: disable=
+            cache: dictionary that carries information between label calls in the
+                same drawing. Required to match general call signature. Not used.
+
+        Returns:
+            str: label to use in drawings
+        """
+        _label = "┤↗ˣ"
+
+        if self.postselect is not None:
+            _label += "₁" if self.postselect == 1 else "₀"
+
+        _label += "├" if not self.reset else "│  │0⟩"
+
+        return _label
+
+    def diagonalizing_gates(self):
+        """Decompose to a diagonalizing gate and a standard MCM in the computational basis"""
+        return [qml.H(self.wires)]
+
+
+class YMidMeasureMP(ParametricMidMeasureMP):
+    """A subclass of ParametricMidMeasureMP that uses the Y measurement basis
+    (angle=pi/2, plane="XY"). For labels and visualizations, this will be represented
+    as a Y measurement. It is otherwise identical to the parent class."""
+
+    _shortname = "measure_y"
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        wires: Optional[Wires],
+        reset: Optional[bool] = False,
+        postselect: Optional[int] = None,
+        id: Optional[str] = None,
+    ):
+        super().__init__(
+            wires=Wires(wires),
+            angle=np.pi / 2,
+            plane="XY",
+            reset=reset,
+            postselect=postselect,
+            id=id,
+        )
+
+    def _flatten(self):
+        metadata = (
+            ("wires", self.raw_wires),
+            ("reset", self.reset),
+            ("id", self.id),
+        )
+        return (None, None), metadata
+
+    def __repr__(self):
+        """Representation of this class."""
+        return f"{self._shortname}(wires={self.wires.tolist()})"
+
+    def label(
+        self, decimals: int = None, base_label: str = None, cache: dict = None
+    ):  # pylint: disable=unused-argument
+        r"""How the mid-circuit measurement is represented in diagrams and drawings.
+
+        Args:
+            decimals: If ``None``, no parameters are included. Else, how to round
+                the parameters. Required to match general call signature. Not used.
+            base_label: overwrite the non-parameter component of the label.
+                Required to match general call signature. Not used.
+            cache: dictionary that carries information between label calls in the
+                same drawing. Required to match general call signature. Not used.
+
+        Returns:
+            str: label to use in drawings
+        """
+        _label = "┤↗ʸ"
+
+        if self.postselect is not None:
+            _label += "₁" if self.postselect == 1 else "₀"
+
+        _label += "├" if not self.reset else "│  │0⟩"
+
+        return _label
+
+    def diagonalizing_gates(self):
+        """Decompose to a diagonalizing gate and a standard MCM in the computational basis"""
+        return [qml.adjoint(qml.S(self.wires)), qml.H(self.wires)]
```

Comment:
```suggestion
        # alternatively we could apply (Z, S) instead of adjoint(S)
        return [qml.adjoint(qml.S(self.wires)), qml.H(self.wires)]
```